lib/matplotlib/scale.py

```diff
@@ -478,10 +478,111 @@ def get_transform(self):
         return self._transform
 
 
+def _mask_non_logit(a):
+    """
+    Return a Numpy masked array where all values outside ]0, 1[ are
+    masked. If all values are inside ]0, 1[, the original array is
+    returned.
+    """
+    a = a.copy()
+    mask = (a <= 0.0) | (a >= 1.0)
+    a[mask] = np.nan
+    return a
+
+
+def _clip_non_logit(a):
```

Comment:
I think you could achieve what you want in one line with `return np.clip(a, 1e-300, 1 - 1e-300)`.  Either way, though, you are assuming `a` is double precision.  Is this a safe assumption?